packages/lib/src/artifacts/ContractFactory.ts

```diff
@@ -1,11 +1,12 @@
 import BN from 'bignumber.js';
 import ZWeb3 from './ZWeb3';
+import Web3 from 'web3';
 import decodeLogs from '../helpers/decodeLogs';
 import { getSolidityLibNames, hasUnlinkedVariables } from '../utils/Bytecode';
 
 interface ContractSchema {
   contractName: string;
-  abi: any;
+  abi: Web3.AbiDefinition[];
```

Comment:
If we update to Web3 1.x, quite a lot of things would change I think. Web3's interface would change, including how it uses promises instead of callbacks. I think that in this case, having some sort of interface would help when that moment comes, because we would be looking for similar structures.

Reply:
I'm concerned about having to modify these custom types because of external decisions (i.e., external lib updates). I prefer leaving all these variables as `any`, but I don't have a strong reason to support my thoughts.